libandroid-navigation/src/main/java/org/maplibre/navigation/android/navigation/v5/models/DirectionsCriteria.kt

```diff
@@ -0,0 +1,316 @@
+package org.maplibre.navigation.android.navigation.v5.models
+
+import androidx.annotation.StringDef
+
+/**
+ * Constants and properties used to customize the directions request.
+ *
+ * @since 1.0.0
+ */
+
+@Suppress("unused")
```

Comment:
What exactly isn't used here? This seems to be pretty Mapbox specific? So I personally wouldn't mind if we remove this object, if it is not used?

Reply:
Most of the parameters are used. As values on the `RouteOptions`, and documentation references.

Only `ANNOTATION_CLOSURE`, `GeometriesCriteria`, `OverviewCriteria`, `SourceCriteria`, `DestinationCriteria` are unused. 

I think we should keep this, until we are making the `RouteOptions` more generic.